### ISSUE TITLE
Git extension detects wrong version over Remote SSH

### ISSUE THREAD

**Issue by randallpittman** [microsoft/vscode#78819]
Same problem as #77607, perhaps.
- VSCode Version:
Version: 1.37.0-insider
Commit: 5e08ee01b7daee99cdda2456f19d77e715fb8762
Date: 2019-08-05T21:25:08.801Z
Electron: 4.2.7
Chrome: 69.0.3497.128
Node.js: 10.11.0
V8: 6.9.427.31-electron.0
OS: Linux x64 4.15.0-55-generic

- OS Version:
    - Local: LinuxMint 19.1 Tessa
    - Remote: CentOS 7.5.1804

- SSH - Remote Extension info:
Name: Remote - SSH
Id: ms-vscode-remote.remote-ssh
Description: Open any folder on a remote machine using SSH and take advantage of VS Code's full feature set.
Version: 0.45.4
Publisher: Microsoft
VS Marketplace Link: https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-ssh

Steps to reproduce:
1. Remote system git version is 1.8.3.1. I don't have admin privileges on this machine.
2. Built git 2.23 on remote system and installed to `~/opt/git`.
Symlinked all binaries to `~/bin`, `~/bin` is ahead of `/bin` on `$PATH`, both vi `~/.bashrc` and `~/.bash_login`.
`git --version` now returns `git version 2.23.0.rc1.10.gc2a8c37`. 
3. Use **Remote - SSH** plugin to connect to project folder on remote host that is a working copy.
4. VS Code notifies me that "You seem to have git 1.8.3.1 installed. Code works best with git >= 2"
5. If I open a terminal in VS Code on the remote host and enter `git --version` I get `git version 2.23.0.rc1.10.gc2a8c37`.
6. If I go to the Git Extension and use the **Show Git Output** option, I get the following:
```
Looking for git in: git
Using git 1.8.3.1 from git
> git rev-parse --show-toplevel
> git rev-parse --git-dir
> git config --get commit.template
Open repository: /nfs/depot/cce_u1/haller/shared/pittmara/Development/sandbox/flask/mega_tutorial
> git status -z -u
> git symbolic-ref --short HEAD
> git rev-parse master
> git rev-parse --symbolic-full-name master@{u}
> git rev-list --left-right master...refs/remotes/gitlab/master
> git for-each-ref --format %(refname) %(objectname) --sort -committerdate
> git remote --verbose
> git check-ignore -v -z --stdin
> git check-ignore -v -z --stdin
> git check-ignore -v -z --stdin
```

**Comment by randallpittman** [comment]
Workaround: set "git.path" in ~/.vscode-server-insiders/data/Machine/settings.json on the remote machine.

**Comment by joaomoreno** [comment]
Good workaround.

Moving to @roblourens for env issues with SSH.

**Comment by roblourens** [comment]
How is git added to your PATH? The server discovers its environment by running an interactive `$SHELL`

**Comment by randallpittman** [comment]
> How is git added to your PATH? The server discovers its environment by running an interactive login `$SHELL`

My `~/.bashrc` file includes the line `export PATH=~/bin:$PATH` at the end, and `~/bin` contains symlinks to the git executables.

Strangely, I cannot reproduce the problem today. I don't know if there was a fix somehow in the latest insiders build or if I fixed some mistake in my bashrc file without realizing it!? 😳 I removed the workaround setting but the correct version of git continues to be detected. I'll close this for now but will let you know if the problem resurfaces.
